app/controllers/transcribe_controller.rb

```diff
@@ -123,113 +123,126 @@ def needs_review
+               @page.status != Page::STATUS_NEEDS_REVIEW &&
+               @page.status != Page::STATUS_INCOMPLETE
+
               @page.update_columns(status: Page::STATUS_INDEXED)
             end
           end
-          @work.work_statistic.recalculate({type: @page.status}) if @work.work_statistic
-          @page.submit_background_processes("transcription")
 
-          #if this is a guest user, force them to sign up after three saves
+          @work.work_statistic&.recalculate({ type: @page.status })
+          @page.submit_background_processes('transcription')
+
+          # if this is a guest user, force them to sign up after three saves
           if current_user.guest?
             deeds = Deed.where(user_id: current_user.id).where(deed_type: DeedType.edited_and_transcribed_pages).count
             if deeds < GUEST_DEED_COUNT
               flash[:notice] = t('.you_may_save_notice', guest_deed_count: GUEST_DEED_COUNT)
             else
-              session[:user_return_to]=collection_transcribe_page_path(@collection.owner, @collection, @work, @page.id)
-              redirect_to new_user_registration_path, :resource => current_user
+              session[:user_return_to] = collection_transcribe_page_path(@collection.owner,
+                                                                         @collection, @work, @page.id)
+              redirect_to new_user_registration_path, resource: current_user
               return
             end
           end
+
           if params[:flow] == 'one-off' && @page.status != Page::STATUS_NEEDS_REVIEW
             redirect_to collection_one_off_list_path(@collection.owner, @collection)
           elsif params[:flow] =~ /user-contributions/ && @page.status != Page::STATUS_NEEDS_REVIEW
             user_slug = params[:flow].sub('user-contributions ', '')
-            redirect_to collection_user_contribution_list_path(@collection.owner, @collection, user_slug)       
+            redirect_to collection_user_contribution_list_path(@collection.owner, @collection, user_slug)
           elsif @quality_sampling
             next_page = @quality_sampling.next_unsampled_page
             if next_page
-              redirect_to collection_sampling_review_page_path(@collection.owner, @collection, @quality_sampling, next_page.id, flow: "quality-sampling")            
+              redirect_to collection_sampling_review_page_path(@collection.owner,
+                                                               @collection, @quality_sampling,
+                                                               next_page.id, flow: 'quality-sampling')
             else
               redirect_to collection_quality_sampling_path(@collection.owner, @collection, @quality_sampling)
             end
           else
-            redirect_to :action => 'assign_categories', page_id: @page.id, collection_id: @collection
+            save_button_clicked = params[:save_to_incomplete] || params[:save_to_needs_review] ||
```

Comment:
Would it make sense to move this variable definition earlier?  Do we need it for anything else?

Reply:
I don't think we need it for anything else right now.